workflow/common/common.go

```diff
@@ -168,6 +168,9 @@ const (
 	// Finalizer to block deletion of the workflow if deletion of artifacts fail for some reason.
 	FinalizerArtifactGC = workflow.WorkflowFullName + "/artifact-gc"
 
+	// Finalizer blocks deletion of pods until we're captured their status.
+	Finalizer = workflow.WorkflowFullName + "/status"
+
```

Comment:
Maybe these 2 finalizers should be named to distinguish what they're being applied on? Like:
```
FinalizerWorkflowArtifactGC
FinalizerPodStatus
```

What do you think?